source/common/router/upstream_request.cc

```diff
@@ -290,11 +274,6 @@ void UpstreamRequest::decodeHeaders(Http::ResponseHeaderMapPtr&& headers, bool e
   }
   stream_info_.response_code_ = static_cast<uint32_t>(response_code);
 
-  if (paused_for_connect_ && response_code == 200) {
```

Comment:
I'm surprised this wasn't guarded by `allow_upstream_filters_` as well.

Reply:
it was implicitly - we set paused_for_connect false when we get response headers from the codec in that path.  that and encodeBodyAndTrailers is a no-op in the new mode anyway